protofsm/daemon_events.go

```diff
@@ -116,6 +116,11 @@ type RegisterConf[Event any] struct {
 	// transaction needs to dispatch an event.
 	NumConfs fn.Option[uint32]
 
+	// FullBlock is a boolean that indicates whether we want the full block
+	// in the returned response. This is useful if callers want to create an
+	// SPV proof for the transaction post conf.
+	FullBlock bool
+
```

Comment:
We could name this field `IncludeBlock` which would align with the `chainntnfs` options field name (`chainntnfs.WithIncludeBlock()` etc).